functions/v2/helloworld/hello_pubsub.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_cloudevent_pubsub]
+
+// Package helloworld provides a set of Cloud Functions samples.
+package helloworld
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+)
+
+// A Message containing the Pub/Sub message.
+// See the documentation for more details:
+// https://cloud.google.com/eventarc/docs/cloudevents#pubsub
+type MessagePublishedData struct {
+	Message PubSubMessage
+}
+
+// PubSubMessage is the payload of a Pub/Sub event.
```

Comment:
suggestion: Referring to both these structs as the message is really confusing. Especially since `MessagePublishedData`"` is the Pub/Sub payload and `PubSubMessage` is specifically the data property. I think even if we diverge from eventarc it worth using clearer naming.

Reply:
I think we should use PubSubMessage cause it's in the documentation here: https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage
And MessagePublishedData seems to be use in the documentation as well.  In this one they call it it `MessagePublished` https://cloud.google.com/eventarc/docs/cloudevents#pubsub